lib/win32/certstore/store_base.rb

```diff
@@ -84,18 +84,23 @@ def cert_list(store_handler)
       # store_handler => Open certificate store handler
       # certificate_thumbprint => thumbprint is a hash. which could be sha1 or md5.
       def cert_delete(store_handler, certificate_thumbprint)
+        validate_thumbprint(certificate_thumbprint)
+        cert_name = memory_ptr
+        thumbprint = update_thumbprint(certificate_thumbprint)
+        cert_pem = get_cert_pem(thumbprint)
+        cert_pem = format_pem(cert_pem)
+        cert_rdn = get_rdn(build_openssl_obj(cert_pem))
+        result = false
         begin
-          if !certificate_name.empty? && (pcert_context = CertFindCertificateInStore(store_handler, X509_ASN_ENCODING, 0, CERT_FIND_ISSUER_STR, certificate_name.to_wstring, nil)) && (not pcert_context.null?)
+          cert_args = cert_find_args(store_handler, cert_rdn)
+          if pcert_context = CertFindCertificateInStore(*cert_args)
             if CertDeleteCertificateFromStore(CertDuplicateCertificateContext(pcert_context))
-              return "Deleted certificate #{certificate_name} successfully"
-            else
-              lookup_error
+              result = true
             end
           end
-          return "Cannot find certificate with name as `#{certificate_name}`. Please re-verify certificate Issuer name or Friendly name"
+          result
```

Comment:
You don't need to set and return values if all we do is a single test. This is to say, these two example are equivalent in Ruby:

```ruby
def example
  result = false
  if true == true
    result = true
  end
  return result
end
```
```ruby
def example_two
  true == true
end
```